Client/Frontend/Browser/TabTrayController.swift

```diff
@@ -223,9 +225,17 @@ struct PrivateModeStrings {
     static let toggleAccessibilityValueOff = NSLocalizedString("Off", tableName: "PrivateBrowsing", comment: "Toggled OFF accessibility value")
 }
 
+protocol TabTrayDelegate {
+    func tabTrayDidDismiss(tabTray: TabTrayController)
+    func addBookmark(tab: Browser)
+    func addToReadingList(tab: Browser) -> ReadingListClientRecord?
+    func present(viewController viewController: UIViewController)
+}
+
 class TabTrayController: UIViewController {
     let tabManager: TabManager
     let profile: Profile
+    var delegate: TabTrayDelegate?
```

Comment:
Nit: `protocol TabTrayDelegate: class` above and `weak var` here to help prevent retain loops.